### ISSUE TITLE
why adam worse than sgd?

### ISSUE THREAD

**Issue by qzfnihao** [mkroutikov/tf-lstm-char-cnn#17]
I tried use adam instead of sgd, but it performance worse. 
Use sgd, in validation data, the default parameters got perplexity 83; 
Use adam, the perplexity is 133; and i tried several parameters about learning_rate,  batch_size, adam learning_rate, but none reduced the perplexity. 
Why?  how to explain this?
In my experience, adam always performance better than sgd in LSTM construction; but it loss the magic in this struction.  Do you have any explaination about this ?

**Comment by mkroutikov** [comment]
I do not have an explanation for that. Looks like an interesting research project for you :)

One possible thing to check is the gradient clipping. It is absolutely needed for SGD. But this can fool Adam, or at least interfere with Adam, as it tries to average gradients... You may want to try turning gradient clipping off...

Good luck and please post here if you find an explanation.
